pkg/orchestration/wiring/datadog/plugin.go

```diff
@@ -0,0 +1,174 @@
+			Message: message,
+			Options: AlarmOptions{
+				EscalationMessage: message,
+				NotifyNoData:      false,
+				RequireFullWindow: true,
+				Thresholds:        query.Threshold,
+			},
+		},
+	}
+
+	serviceInstanceSpecBytes, err := json.Marshal(&serviceInstanceSpec)
+	if err != nil {
+		return smith_v1.Resource{}, errors.WithStack(err)
+	}
+
+	alarmInstanceResource := smith_v1.Resource{
+		Name: wiringutil.ConsumerProducerResourceNameWithPostfix(resource.Name, kubeCompute, string(alarmType)),
+		References: []smith_v1.Reference{
+			{
+				Resource: kubeDeployment.Data.DeploymentResourceName,
+			},
+		},
+		Spec: smith_v1.ResourceSpec{
+			Object: &sc_v1b1.ServiceInstance{
+				TypeMeta: meta_v1.TypeMeta{
+					Kind:       "ServiceInstance",
+					APIVersion: sc_v1b1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name: wiringutil.ConsumerProducerMetaNameWithPostfix(resource.Name, kubeCompute, string(alarmType)),
+				},
+				Spec: sc_v1b1.ServiceInstanceSpec{
+					PlanReference: sc_v1b1.PlanReference{
+						ClusterServiceClassExternalID: clusterServiceClassExternalID,
+						ClusterServicePlanExternalID:  clusterServicePlanExternalID,
+					},
+					Parameters: &runtime.RawExtension{
+						Raw: serviceInstanceSpecBytes,
+					},
+					ExternalID: instanceID,
+				},
+			},
+		},
+	}
+	return alarmInstanceResource, nil
+}
+
+func validateRequest(stateResource *orch_v1.StateResource, context *wiringplugin.WiringContext) error {
+	if stateResource.Type != ResourceType {
+		return errors.Errorf("invalid resource type: %q", stateResource.Type)
+	}
+	if stateResource.Spec != nil {
+		return errors.New("default alarm does not accept any user parameters")
+	}
+	return nil
+}
+
+func (q *QueryParams) generateQuery() string {
+	if q.AlarmType == CPU {
+		cpuUsageString := fmt.Sprintf("avg(last_5m):( avg:kubernetes.cpu.usage.total{env:%s,kube_namespace:%s,kube_deployment:%s,region:%s} by {container_id} ", q.Location.EnvType, q.KubeNamespace, q.KubeDeployment, q.Location.Region)
+		cpuLimitString := fmt.Sprintf("/ ( avg:kubernetes.cpu.limits{env:%s,kube_namespace:%s,kube_deployment:%s,region:%s} by {container_id} * 1000000 ) ) * 100 > %d", q.Location.EnvType, q.KubeNamespace, q.KubeDeployment, q.Location.Region, q.Threshold.Critical)
+		return cpuUsageString + cpuLimitString
+	}
+
+	memoryUsageString := fmt.Sprintf("avg(last_5m):( avg:kubernetes.memory.usage.total{env:%s,kube_namespace:%s,kube_deployment:%s,region:%s} by {container_id} ", q.Location.EnvType, q.KubeNamespace, q.KubeDeployment, q.Location.Region)
+	memoryLimitString := fmt.Sprintf("/ ( avg:kubernetes.memory.limits{env:%s,kube_namespace:%s,kube_deployment:%s,region:%s} by {container_id} * 1000000 ) ) * 100 > %d", q.Location.EnvType, q.KubeNamespace, q.KubeDeployment, q.Location.Region, q.Threshold.Critical)
+	return memoryUsageString + memoryLimitString
+
+}
+
+func (q *QueryParams) generateMessage(notificationProp *meta_orch.Notifications) string {
```

Comment:
Label is not handled

Reply:
Namespace or label should be included in the message, otherwise it is not clear which deployment the alert is caused by.